internal/cli/cmd/migrate/migrate.go

```diff
@@ -154,6 +159,22 @@ func (o *option) Execute() error {
 		return err
 	}
 	persesDashboard.Metadata.Project = o.project
+
+	// Apply datasource cleaning if the flag is set
+	if o.useDefaultDatasource {
```

Comment:
I think this should go in the migration logic directly.

Then you will have to expose a query parameter in the migrate endpoint for the online migration. For the offline migration you will have to call the migrate method with this parameter